atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepDelay.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.queue;
+
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class SweepDelay {
+    public static final int BATCH_CELLS_LOW_THRESHOLD = 100;
+    public static final long MIN_PAUSE_MILLIS = 1;
+    public static final long MAX_PAUSE_MILLIS = 5000;
+    public static final long BACKOFF = Duration.ofMinutes(2).toMillis();
+
+    private final long initialPause;
+    private final AtomicLong currentPause;
+
+    public SweepDelay(long configPause) {
+        this.initialPause = Math.min(Math.max(MIN_PAUSE_MILLIS, configPause), MAX_PAUSE_MILLIS);
```

Comment:
I'd question whether we want to place a cap on MAX_PAUSE_MILLIS: it seems a bit unexpected that if the user configures the gap to be 10 seconds for whatever reason we just throw it away. Even if we think that's ok I think it makes sense to fail more aggressively.

In other words I'd go for 
```
initial = max(min, config)
min = 1
max = max(5000, config)
```

Note that if you go this route you may need some special integer mathematics in line 47...

Reply:
Makes sense, I now just use it as both initial and max. Do we want to do something like max should always be at least 2 * initial instead?